src/native/libs/System.Globalization.Native/pal_locale.m

```diff
@@ -0,0 +1,30 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include <stdlib.h>
+#include "pal_locale_internal.h"
+
+#import <Foundation/Foundation.h>
+
+char* DetectDefaultAppleLocaleName()
+{
+    NSLocale *currentLocale = [NSLocale currentLocale];
+    NSString *localeName = @"";
+    
+    if (!currentLocale)
+    {
+        return strdup([localeName UTF8String]);
+    }
+
+    if ([currentLocale.languageCode length] > 0 && [currentLocale.countryCode length] > 0)
+    {
+        localeName = [NSString stringWithFormat:@"%@-%@", currentLocale.languageCode, currentLocale.countryCode];
+    }
+    else
+    {
+        localeName = [currentLocale.localeIdentifier stringByReplacingOccurrencesOfString:@"_" withString:@"-"];
```

Comment:
The docs say it could, but for the life of me I can't get the current locale to include it.  Should I just take the string before `@collation`?

Reply:
I guess you may be able to get the current locale with the collation if use `de-DE` culture. I am not sure if there will be a UI setting that lets you select the sorting order (like phonebook or something similar).    

When having a locale with a collation, something like `de_DE@collation=phoneb` this will be translated to .NET locale name as `de-DE_phoneb`.

Reply:
@tarekgh No matter what language and list sort order I choose, I still cannot produce a `localeIdentifier` with a collation value in it.  

@rolfbjarne do you happen to know the magic settings?

Reply:
@steveisok there are examples here: https://developer.apple.com/documentation/foundation/nslocale/1416263-localeidentifier?language=objc: `Examples of locale identifiers include "en_GB", "es_ES_PREEURO", and "zh-Hant_HK_POSIX@collation=pinyin;currency=CNY".`

Although the documentation here doesn't say anything about collation: https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPInternational/LanguageandLocaleIDs/LanguageandLocaleIDs.html#//apple_ref/doc/uid/10000171i-CH15-SW9

If it's for testing, you could iterate over all locale identifier and see what's available: https://developer.apple.com/documentation/foundation/nslocale/1410448-availablelocaleidentifiers?language=objc

Alternatively you could create a locale with whatever identifier you want to see if you get the same thing back: https://developer.apple.com/documentation/foundation/nslocale/1414217-initwithlocaleidentifier?language=objc

Reply:
I can definitely create it myself, but if the you can't from the settings, I'm not sure we need to provide any transformation.

Reply:
I tried a couple of string combos and what it tells me is we're processing the collation correctly.

`zh-Hant_HK_POSIX@collation=pinyin;currency=CNY` turns out to be `zh-Hant-HK-POSIX_pinyin`

`de_DE@collation=phoneb` turns out to be `de_DE`

@tarekgh I don't think we'll need any changes as https://github.com/steveisok/runtime/blob/5f367d5fcad90916cdb94531bff4955fb63b5a40/src/native/libs/System.Globalization.Native/pal_locale.c#L243-L251 already transforms.

Reply:
> I don't think we'll need any changes as [steveisok/runtime@5f367d5/src/native/libs/System.Globalization.Native/pal_locale.c#L243-L251](https://github.com/steveisok/runtime/blob/5f367d5fcad90916cdb94531bff4955fb63b5a40/src/native/libs/System.Globalization.Native/pal_locale.c#L243-L251) already transforms.

Right, Now in your new code you need to remove the normalization you have done I guess (which replacing `_` with `-`).